securedrop_client/api_jobs/sync.py

```diff
@@ -37,11 +38,38 @@ def call_api(self, api_client: API, session: Session) -> Any:
         # This timeout is used for 3 different requests: `get_sources`, `get_all_submissions`, and
         # `get_all_replies`
         api_client.default_request_timeout = 60
+        users = api_client.get_users()
+        MetadataSyncJob._update_users(session, users)
         sources, submissions, replies = get_remote_data(api_client)
-
         update_local_storage(session, sources, submissions, replies, self.data_dir)
-        user = api_client.get_current_user()
-        if "uuid" in user and "username" in user and "first_name" in user and "last_name" in user:
-            create_or_update_user(
-                user["uuid"], user["username"], user["first_name"], user["last_name"], session
-            )
+
+    def _update_users(session: Session, remote_users: List[User]) -> None:
```

Comment:
Overall this seems like a good candidate for `session.merge()`: https://docs.sqlalchemy.org/en/13/orm/session_state_management.html#unitofwork-merging

That said I've never used it before (didn't have a use case until now!) so if it's not already being used in SD, we could leave it as a TODO for the future.